craft/capctl/src/capctl.scala

```diff
@@ -57,43 +58,33 @@ class NcapctlTop(c: NcapctlTopParams)(implicit p: Parameters) extends NcapctlTop
     ioBridgeSink.bundle.reset_n := !(reset.toBool)
 
     // connect ioBridge source and sink
-    ioBridgeSource.bundle.odata   := ioBridgeSink.bundle.odata
+
+    ioNode.bundle.odata := ioBridgeSink.bundle.odata
   }
 
 }
 
 object NcapctlTop {
   def attach(c: NcapctlTopParams)(bap: BlockAttachParams): NcapctlTop = {
     val capctl = NcapctlTopBase.attach(c)(bap)
+    val pwm = NPWMTopBase.attach(NPWMTopParams.defaults(0x70000L, c.blackbox.cacheBlockBytes))(bap)
```

Comment:
you were missing the `PWM` module instantiation and attachment. you need to call the `NPWMTopBase.attach` method to instantiate a `NPWMTop` module and also attach it's apb interface.